src/main/java/unbanner/BuildingController.java

```diff
@@ -35,42 +38,76 @@ public String newBuilding(@ModelAttribute("building") Building building) {
     return "redirect:/buildings";
   }
 
+  //Get building
   @RequestMapping("/building/{id}")
   public String room0(@PathVariable String id, Model model) {
     model.addAttribute("building", repository.findOne(id));
     return "building";
   }
 
+  //Delete building
   @RequestMapping(value = "/building/{id}", method = RequestMethod.DELETE)
   public String room0(@PathVariable String id) {
     repository.delete(id);
     return "redirect:/buildings";
   }
 
+  //Update building
   @RequestMapping(value = "/building/{id}", method = RequestMethod.POST)
-  public String room0(@ModelAttribute("building") Building building) {
-    repository.save(building);
+  public String room0(@ModelAttribute("building") Building building,
+                      @PathVariable String id) {
+    Building tempBld = repository.findById(id);
+    tempBld.name = building.name;
+    tempBld.description = building.description;
+    repository.save(tempBld);
     return "redirect:/buildings";
   }
 
-
+  //Get room
   @RequestMapping("/building/room/{id}")
   public String room(@PathVariable String id, Model model) {
     model.addAttribute("room", roomRepository.findOne(id));
     return "room";
   }
 
+  //Delete room
   @RequestMapping(value = "/building/room/{id}", method = RequestMethod.DELETE)
   public String room(@PathVariable String id) {
     roomRepository.delete(id);
     return "redirect:/buildings";
   }
-
+  //Update room
   @RequestMapping(value = "/building/room/{id}", method = RequestMethod.POST)
-  public String room(@ModelAttribute("room") Room room) {
-    roomRepository.save(room);
+  public String room(@ModelAttribute("room") Room room,
+                     @PathVariable String id){
+    Room tempRoom = roomRepository.findById(id);
+    tempRoom.name = room.name;
+    tempRoom.size = room.size;
+    roomRepository.save(tempRoom);
     return "redirect:/buildings";
   }
 
 
+  //Get building for create_room
+  @RequestMapping("/building/newRoom/{id}")
+  public String newroom0(@PathVariable String id, Model model) {
+    model.addAttribute("building", repository.findOne(id));
+    return "create_room";
+  }
+
+  //Create room
+  @RequestMapping(value = "/buildings/newRoom/{id}", method = RequestMethod.POST)
+  public String newRoom(@ModelAttribute("room") Building bld,
+                        @PathVariable String id) {
+    Building thisBuilding = repository.findById(id);
+    Room newRoom = new Room();
+    newRoom.name ="New Room";
```

Comment:
CheckStyle Issue: WhitespaceAround: '=' is not followed by whitespace. Empty blocks may only be represented as {} when not part of a multi-block statement (4.1.3)